src/WebJobs.Extensions/Extensions/Core/Bindings/ExecutionContextBindingProvider.cs

```diff
@@ -60,17 +71,24 @@ public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
                     throw new ArgumentNullException("context");
                 }
 
-                return BindInternalAsync(CreateContext(context.FunctionInstanceId));
+                return BindInternalAsync(CreateContext(context));
             }
 
-            private ExecutionContext CreateContext(Guid invocationId)
+            private ExecutionContext CreateContext(ValueBindingContext context)
             {
-                return new ExecutionContext
+                var result = new ExecutionContext
                 {
-                    InvocationId = invocationId,
-                    FunctionName = _parameter.Member.Name,
-                    FunctionDirectory = Environment.CurrentDirectory
+                    InvocationId = context.FunctionInstanceId,
+                    FunctionName = context.FunctionContext.MethodName,
+                    FunctionDirectory = Environment.CurrentDirectory,
```

Comment:
Seems quirky to rely on Current Dir, as that's fairly random in IIS scenarios (e.g. I think it's system32 folder). Though I take it that in real scenarios, the code below that overrides it will always run, making this one moot?